### DISCUSSION TITLE
OMPL for under-actuated robotic arms

### DISCUSSION THREAD

**Discussion by DentOpt** [moveit/moveit2#2709] [Q&A]
I am currently integrating a 3DOF robotic arm. 
Accordingly I would like to have a control only of the arm orientation.
If I give the full pose, there is evidently no solution found as the problem is over-constraint. 
I am currently using ompl with the following C++ interface:

```
    geometry_msgs::msg::PoseStamped goal;
    goal.header.frame_id="world";
    goal.pose.position.x = 0;
    goal.pose.position.y = 0;
    goal.pose.position.z = 0;
    goal.pose.orientation.x = 1;
    goal.pose.orientation.y = 0;
    goal.pose.orientation.z = 0;
    goal.pose.orientation.w = 0;
    goal.header.frame_id="world";

    arm_->setGoal(goal, std::string("end_effector"));
    const auto plan_solution = arm_->plan();
    if (plan_solution)
    {
      arm_->execute();
    }
```

Is there a smart way to configure ompl to only track specific axis/rotations (e.g. x,y,yaw) according to the degrees of freedom of the arm?

Thanks

**Comment by mamoll** [comment]
You could set the position/angle tolerance for the goal with [constructGoalConstraints](https://github.com/ros-planning/moveit2/blob/b4ff391133c2809e9f697d44593c89a77d1d4c5c/moveit_core/kinematic_constraints/include/moveit/kinematic_constraints/utils.h#L139) to indicate that error in certain dimensions can be ignored. A constraint generated with this helper function can then be used to set the goal via the moveit_cpp or MoveGroup interface.
